src/main/pages/setup-docker/docker-config.adoc

```diff
@@ -171,60 +172,68 @@ docker run <OTHER-DOCKER_OPTIONS> -e CHE_HOST=<ip-addr-or-dns> eclipse/che:<vers
-#    1. If set, use value of CHE_INFRA_DOCKER_MASTER__API__ENDPOINT
-#    2. Default is 'http://che-host:${SERVER_PORT}/api', where 'che-host' is IP of docker0 (linux) VM IP (Mac and Win).
-----
+* `CHE_API`: Points to the same value as `CHE_API_INTERNAL` for backward compatibility.
+
+[NOTE]
+====
+The `CHE_API` environment variable will be removed in a future release.
+====
+
+When it is booting or starting a workspace, Che goes through a progression algorithm to establish the protocol, IP address, and port to establish communication. You can override certain parameters in Che’s configuration to overcome issues with the Docker daemon, workspaces, or browsers being on different networks.
+
+*Browser -> Che server*
+
+The '${CHE_HOST}:${SERVER_PORT}/wsmaster/api' variable is the default. However, requests are sent to the IP/hostname in your browser's address bar. You can also use the value of `che.api`.
+
+*Che server -> Docker daemon progression*
+
+Use the value of the `che.infra.docker.daemon_url` variable. You can also use one of the following values:
+
+* The value of the `DOCKER_HOST` system variable
 
-It is common for configuration with firewalls, routers, networks and hosts to make the default values we detect to establish these connections incorrect. You can run `docker run <DOCKER_OPTIONS> eclipse/che info --network` to run a test that makes connections between simulated components to reflect the networking setup of Che as it is configured. You do not need all connections to pass for Che to be properly configured. For example, on a Windows machine, this output may exist, just indicating that `localhost` is not an acceptable domain for communications, but the IP address `10.0.75.2` is.
+* The Unix socket over `unix:///var/run/docker.sock`
+
+* The default `docker0 IP` - 172.17.42.1
+
+*Che Server -> Workspace Connection*
+
+Use the value of the `che.docker.ip` variable. Or, use the address of the docker0 bridge network, if available.
+
+*Browser -> Workspace Connection*
+
+Use the value of the `che.docker.ip.external` variable. Or, use the `che.docker.ip` value, or the value provided by ws container inspect.
+
+*Workspace Agent -> Che Server*
+
+If set, use the value of the `CHE_INFRA_DOCKER_MASTER__API__ENDPOINT` variable. The default value is 'http://che-host:${SERVER_PORT}/api'; here, 'che-host' is the IP of docker0 (linux) or the VM IP (macOS and Windows).
+
+The configuration with firewalls, routers, networks, and hosts makes the detected default values to establish these connections incorrect. You can run the `docker run <DOCKER_OPTIONS> eclipse/che info --network` command to run a test that makes connections between simulated components to reflect the networking setup of Che as it is configured. All connections may not pass for Che to be properly configured. For example, on a Windows machine, this output may exist indicating that `localhost` is not an acceptable domain for communication but the IP address `10.0.75.2` is.
```

Comment:
It did not. Changed to "Che's configuration with firewalls, routers, networks, and hosts makes the detected default values to establish faulty connections." Does that make sense?

Reply:
Without the 'to', it  would make sense grammatically. I still don't understand the meaning. You're saying that Che config creates faulty connections. Why? What specific config? How faulty? How to prevent it? The para is weird. For example, what's the meaning of _"All connections may not pass for Che to be properly configured."_?

Btw, same mistake with 'to' in the 2nd sentence. Correct use is "it makes me do something", not "it makes me to do something".

Reply:
@rkratky , I asked Shoubhik from the Dev team and he said that it is _okay_ to have the faulty connections and that is exactly what we are trying t tell the user in this paragraph. 

To make it clearer, I changed it to "When Che is configured with firewalls, routers, networks, and hosts, the default values detected, as a known behavior, will establish faulty connections. Execute the `docker run <DOCKER_OPTIONS> eclipse/che info --network` command to run a test that makes connections between simulated components reflect the networking setup of Che as it is configured. All connections may not pass for Che to be properly configured and that is acceptable. For example, on a Windows machine, this output may exist indicating that `localhost` is not an acceptable domain for communication but the IP address `10.0.75.2` is."

Does that make sense?